pkg/test/syncset/syncset.go

```diff
@@ -0,0 +1,95 @@
+package syncset
+
+import (
+	hivev1 "github.com/openshift/hive/pkg/apis/hive/v1alpha1"
+	controllerutils "github.com/openshift/hive/pkg/controller/utils"
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+// Option defines a function signature for any function that wants to be passed into BuildSyncSet
+type Option func(*hivev1.SyncSet)
+
+// ChecksumFunc defines a function signature for checksumming objects.
+type ChecksumFunc func(runtime.Object) (string, error)
+
+// Build runs each of the functions passed in to generate a cluster deployment.
+func Build(opts ...Option) *hivev1.SyncSet {
+	retval := &hivev1.SyncSet{}
+	for _, o := range opts {
+		o(retval)
+	}
+
+	return retval
+}
+
+// WithName sets the SyncSet.Name field when building a syncset with BuildSyncSet.
+func WithName(name string) Option {
```

Comment:
We should consider making some of these common options available in a generic way instead of needing to copy these for each specific type.

In the pkg/test/generic package,
```
type Option func(obj metav1.Object)

func WithName(name string) Option {
  return func(obj metav1.Object) {
    obj.SetName(name)
  }
}
```
In, for example, pkg/test/syncset
```
func Generic(opt generic.Option) Option {
  return func(syncSet *hivev1.SyncSet) {
    opt(syncSet)
  }
}
```
To use,
```
syncset.Build(syncset.Generic(generic.WithName("test-name")))
```

It makes the caller code a bit more dense at the benefit of not having to copy blocks of code in the callee side.